pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -455,18 +464,19 @@ type ConsulSDConfig struct {
 	// If unset, Prometheus uses its default value.
 	// +optional
 	RefreshInterval *v1.Duration `json:"refreshInterval,omitempty"`
-	// BasicAuth information to authenticate against the Consul Server.
+	// Optional BasicAuth information to authenticate against the Consul Server.
 	// More info: https://prometheus.io/docs/operating/configuration/#endpoints
+	// Cannot be set at the same time as `authorization`, or `oauth2`.
 	// +optional
 	BasicAuth *v1.BasicAuth `json:"basicAuth,omitempty"`
-	// Authorization header configuration to authenticate against the Consul Server.
+	// Optional Authorization header configuration to authenticate against the Consul Server.
+	// Cannot be set at the same time as `basicAuth`, or `oauth2`.
```

Comment:
may we should explore the cel way to validate the mutual exclusion here can be done in follow-up PR as well
https://kubernetes.io/docs/reference/using-api/cel/

Reply:
Could we do this better by grouping the authentication mechanisms under something like `HTTPClientConfig` or something like that? that would save a lot of code duplication and margin for error.

Reply:
Logged it as an issue. I will resolve this comment.